code/modules/reagents/chemistry/reagents/medicine_reagents.dm

```diff
@@ -875,16 +875,37 @@
 
 /datum/reagent/medicine/mutadone
 	name = "Mutadone"
-	description = "Removes jitteriness and restores genetic defects."
+	description = "Removes jitteriness and temporarly supresses genes from manifesting, it also prevents new mutation from appearing while in the system."
 	color = "#5096C8"
 	taste_description = "acid"
+	overdose_threshold = 35
+	var/list/mut = list()
+
+/datum/reagent/medicine/mutadone/on_mob_metabolize(mob/living/carbon/M)
+	..()
+	var/len = M.dna.mutations.len
+	for(var/i in 1 to len)
```

Comment:
M.dna.mutations.len

assuming all carbons have/use DNA. Pretty sure there's a check specifically for that.

Reply:
FYI you can also do

```DM
for (var/i in M.dna.mutations)
    var/datum/mutation/beep = i
    mut += beep.type
```

Just seems weird to convert to length then convert back to what the thing actually is when you can just loop through entries directly imo.